### DISCUSSION TITLE
Problems with ProxyAgent and Docker

### DISCUSSION THREAD

**Discussion by MinerPL** [nodejs/undici#3438] [Q&A]
I have an application running in production on the docker and in development time on the host system.
On production there are issues with ProxyAgent being used for all queries instead of those with a designated dispatcher.

Code to create ProxyAgent:
```js
async createAgent() {
		const uri = 'http://IP:PORT';
		const token = 'Basic ' + Buffer.from('LOGIN:PASSWORD').toString('base64');
		const agent = new ProxyAgent({ uri, token });
		setGlobalDispatcher(new Agent());

		fetch('https://ifconfig.me/ip', {
			dispatcher: agent,
		}).then(() => {
			this.agent = agent;
		}).catch(e => {
			console.log(e);
		});

		return this.agent;
	}
```

The code I used to ensure that both requests use ProxyAgent.
```js
await fetch('https://ifconfig.me/ip').then(r => r.text()) // Return Proxy IP

await fetch('https://ifconfig.me/ip', {dispatcher: client.proxy.getProxyAgent()}).then(r => r.text()) // Return Proxy IP
```

**Comment by mcollina** [comment]
I don't really understand what the problem is.
Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.
